src/bin/miri.rs

```diff
@@ -749,6 +744,29 @@ fn main() {
     let many_seeds =
         many_seeds.map(|seeds| ManySeedsConfig { seeds, keep_going: many_seeds_keep_going });
 
+    // Validate settings for data race detection and GenMC mode.
+    match (
```

Comment:
I think this would become easier to read if you did a top-level `if miri_config.genmc_config.is_some()` rahter than cramming it all into a single `match`.

I assume you cannot disable weak_memory_emulation in GenMC mode either.